third-party/realdds/src/dds-topic.cpp

```diff
@@ -18,13 +18,22 @@ dds_topic::dds_topic( std::shared_ptr< dds_participant > const & participant,
     : _participant( participant )
     , _topic( nullptr )
 {
-    // REGISTER THE TYPE
+    // Auto fill DDS X-Types TypeObject so other applications (e.g sniffer) can dynamically match a reader for this topic
+    topic_type.get()->auto_fill_type_object( true );
+    // Don't fill DDS X-Types TypeInformation, it is wasteful if you send TypeObject anyway
+    topic_type.get()->auto_fill_type_information( false );
+    // Registering the topic type with the participant enables topic instance creation by factory
     DDS_API_CALL( topic_type.register_type( _participant->get() ) );
 
-    // CREATE THE TOPIC
+    // Create the topic
     _topic = DDS_API_CALL( _participant->get()->create_topic( topic_name,
                                                               topic_type->getName(),
                                                               eprosima::fastdds::dds::TOPIC_QOS_DEFAULT ) );
+
+    // Topic constructor creates TypeObject that will be sent as part of DDS discovery phase
+    void * data = topic_type->createData();
```

Comment:
Will this also work for topics that have no type info?

Reply:
Yes, type_support can create data even without type-object information. The type-object support just adds a line in the constructor.